src/Engine/GameEvents/CreateUnitEvent.java

```diff
@@ -60,7 +70,27 @@ public void performEvent(MapMaster gameMap)
     if( null != myNewUnit )
     {
       myCommander.units.add(myNewUnit);
-      gameMap.addNewUnit(myNewUnit, myBuildCoords.xCoord, myBuildCoords.yCoord, myBoots);
+      if( myFudgeRadius < 1 || myBoots )
+        gameMap.addNewUnit(myNewUnit, myBuildCoords.xCoord, myBuildCoords.yCoord, myBoots);
+      else
+      {
+        boolean success = false;
+        // Check concentric rings outwards, to spawn as close to the original spot as possible
+        for( int radius = 0; !success && radius <= myFudgeRadius; ++radius )
+          for( XYCoord xyc : Utils.findLocationsInRange(gameMap, myBuildCoords, radius, radius) )
+          {
+            Unit resident = gameMap.getResident(xyc);
+            if( resident == null )
+            {
+              gameMap.addNewUnit(myNewUnit, xyc.xCoord, xyc.yCoord, false);
+              success = true;
+              break;
```

Comment:
Shouldn't this check for traversability as well?

Reply:
Uh, probably yes.
That does bring to my attention that the animation for "fuzzy spawns" can be wrong, though I'm not hugely worried about it?
I don't see an obvious way to guarantee (by construction) that the animation and actual event go to the same place.

You can kinda assume it'll work if you just update the destination coordinate after the animation, probably, but that turns a getter into a secret setter.
Recalculating should work, assuming the iteration ordering is the same.

Calculating the actual drop point at event creation time doesn't work because other events can happen between creation and execution.

Reply:
> You can kinda assume it'll work if you just update the destination coordinate after the animation, probably, but that turns a getter into a secret setter.

You mean, recalculating the spawn location in the animation right before actually drawing it, and then updating the event with that result as well? Wouldn't work for previews, of course, but again, maybe not the end of the world.